gpytorch/kernels/hamming.py

```diff
@@ -0,0 +1,150 @@
+from typing import Optional
+
+import torch
+from torch import nn, Tensor
+
+from ..constraints import Interval, Positive
+from ..kernels import Kernel
+from ..priors import Prior
```

Comment:
If you import from the specific module rather than the `__all__` defined in `__init__.py` you should avoid circular import issues here. 
```suggestion
from gpytorch.constraints.constraints import Interval, Positive
from gpytorch.kernels.kernel import Kernel
from gpytorch.priors.prior import Prior
```